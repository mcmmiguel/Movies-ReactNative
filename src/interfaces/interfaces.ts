import { Dispatch, SetStateAction } from 'react';
import { Cast } from './creditsInterface';
import { Movie, MovieFull } from './movieInterface';

export interface CastItemProps {
    actor: Cast
}

export interface GradientBackgroundProps {
    children: JSX.Element | JSX.Element[];
}

export interface HorizontalSliderProps {
    title?: string;
    movies: Movie[];
}

export interface MovieDetailsProps {
    movieFull: MovieFull;
    cast: Cast[];
}

export interface MoviePosterProps {
    movie: Movie;
    height?: number;
    width?: number;
}

export interface MovieDetails {
    isLoading: boolean;
    movieFull?: MovieFull;
    cast: Cast[];
}

export interface MoviesState {
    nowPlaying: Movie[];
    popular: Movie[];
    topRated: Movie[];
    upcoming: Movie[];
}

export interface SearchMovieResultsProps {
    movieResults: Movie[];
}

export interface SearchBarProps {
    searchQuery: string;
    setSearchQuery: Dispatch<SetStateAction<string>>;
    setSearchResults: Dispatch<SetStateAction<Movie[]>>;
}
