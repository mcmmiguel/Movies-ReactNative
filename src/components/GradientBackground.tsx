import React from 'react';
import { View, StyleSheet } from 'react-native';
import LinearGradient from 'react-native-linear-gradient';

interface GradientBackgroundProps {
    children: JSX.Element | JSX.Element[];
}

export const GradientBackground = ({ children }: GradientBackgroundProps) => {
    return (
        <View style={{
            flex: 1,
            backgroundColor: '#084F6A',
        }}>
            <LinearGradient
                colors={['#084f6a', '#75cedb', 'white']}
                style={{ ...StyleSheet.absoluteFillObject }}
                start={{ x: 0.1, y: 0.1 }}
                end={{ x: 0.5, y: 0.5 }}
            />
            {children}
        </View>
    );
};
