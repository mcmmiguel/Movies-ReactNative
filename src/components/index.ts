export { MoviePoster } from './MoviePoster';
export { HorizontalSlider } from './HorizontalSlider';
export { MovieDetails } from './MovieDetails';
