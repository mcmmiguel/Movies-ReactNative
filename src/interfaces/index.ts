export * from './movieInterface';
export * from './creditsInterface';
export * from './interfaces';
export * from './searchInterface';
