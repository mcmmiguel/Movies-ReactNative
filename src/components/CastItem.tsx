import React from 'react';
import { View, Text, Image, StyleSheet } from 'react-native';
import { CastItemProps } from '../interfaces';

export const CastItem = ({ actor }: CastItemProps) => {

    const uri = `https://image.tmdb.org/t/p/w500${actor.profile_path}`;

    return (
        <View style={styles.container}>
            {actor.profile_path &&
                <Image
                    testID="actor-image"
                    source={{ uri }}
                    style={styles.image}
                />
            }
            <View style={styles.actorInfo}>
                <Text style={styles.actorName}>
                    {actor.name}
                </Text>
                <Text style={styles.character}>
                    {actor.character}
                </Text>
            </View>
        </View>
    );
};

const styles = StyleSheet.create({
    container: {
        flexDirection: 'row',
        backgroundColor: '#E1E5F2',
        alignItems: 'center',
        height: 60,
        borderRadius: 10,
        shadowColor: '#000',
        shadowOffset: {
            width: 0,
            height: 3,
        },
        shadowOpacity: 0.29,
        shadowRadius: 4.65,

        elevation: 9,

        marginRight: 20,
        paddingVertical: 10,
        paddingRight: 10,
    },
    image: {
        width: 60,
        height: 60,
        borderRadius: 10,
        resizeMode: 'contain',
    },
    actorInfo: {
        justifyContent: 'center',
        marginLeft: 10,
        marginVertical: 4,
    },
    actorName: {
        fontSize: 18,
        fontWeight: 'bold',
        color: 'black',
    },
    character: {
        fontSize: 14,
        color: 'black',
    },
});
