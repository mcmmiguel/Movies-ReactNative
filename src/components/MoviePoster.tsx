import React from 'react';
import { View, Image, StyleSheet } from 'react-native';
import { Movie } from '../interfaces/movieInterface';

interface MoviePosterProps {
    movie: Movie;
    height?: number;
    width?: number;
}

export const MoviePoster = ({ movie, height = 420, width = 300 }: MoviePosterProps) => {

    const uri = `https://image.tmdb.org/t/p/w500${movie.poster_path}`;

    return (
        <View style={[{ height, width }, styles.mainContainer]}>
            <View style={styles.imageContainer}>

                <Image
                    source={{ uri }}
                    style={styles.image}
                />
            </View>
        </View>
    );
};

const styles = StyleSheet.create({
    mainContainer: {
        marginHorizontal: 7.5,
    },
    imageContainer: {
        flex: 1,
        borderRadius: 18,
        shadowColor: '#000',
        shadowOffset: {
            width: 0,
            height: 2,
        },
        shadowOpacity: 0.5,
        shadowRadius: 3.84,

        elevation: 5,
    },
    image: {
        flex: 1,
        borderRadius: 18,
    },
});
