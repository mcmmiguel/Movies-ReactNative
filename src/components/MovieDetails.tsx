import React from 'react';
import { View, Text, StyleSheet } from 'react-native';
import { Cast, MovieFull } from '../interfaces';
import Icon from 'react-native-vector-icons/Ionicons';
import currencyFormatter from 'currency-formatter';
import { globalStyles } from '../../styles';
import { CastItem } from './CastItem';
import { FlatList } from 'react-native-gesture-handler';

interface MovieDetailsProps {
    movieFull: MovieFull;
    cast: Cast[];
}

export const MovieDetails = ({ movieFull, cast }: MovieDetailsProps) => {
    return (
        <>
            {/* Details */}
            <View style={styles.mainContainer}>
                <View style={styles.rateContainer}>
                    <Icon
                        name="star"
                        color="yellow"
                        size={16}
                    />
                    <Text style={styles.voteAverage}>{movieFull.vote_average}</Text>
                    <Text style={styles.genresText}>
                        - {movieFull.genres.map(g => g.name).join(', ')}
                    </Text>
                </View>
            </View>

            {/* Story */}
            <Text style={styles.subtitle}>Sinopsis</Text>
            <Text style={styles.overview}>{movieFull.overview}</Text>

            {/* Budget */}
            <Text style={styles.subtitle}>Budget</Text>
            <Text style={globalStyles.colorText}>{currencyFormatter.format(movieFull.budget, { code: 'USD' })} USD</Text>

            {/* Cast*/}
            <View style={styles.castContainer}>
                <Text style={styles.subtitle}>Actors</Text>
                <FlatList
                    horizontal
                    showsHorizontalScrollIndicator={false}
                    data={cast}
                    keyExtractor={(item) => item.id.toString()}
                    renderItem={({ item }) => <CastItem actor={item} />}
                    style={{ marginTop: 10 }}
                />
            </View>
        </>
    );
};

const styles = StyleSheet.create({
    mainContainer: {
    },
    rateContainer: {
        flexDirection: 'row',
    },
    voteAverage: {
        color: 'black',
    },
    genresText: {
        marginLeft: 5,
        color: 'black',
    },
    subtitle: {
        fontSize: 25,
        marginTop: 10,
        fontWeight: 'bold',
        color: '#212121',
    },
    overview: {
        color: 'grey',
        fontSize: 16,
    },
    castContainer: {
        marginTop: 10,
        marginBottom: 100,
    },
});
