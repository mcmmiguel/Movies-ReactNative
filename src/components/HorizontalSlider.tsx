import React from 'react';
import { View, Text, StyleSheet, FlatList } from 'react-native';
import { MoviePoster } from './MoviePoster';
import { HorizontalSliderProps } from '../interfaces';

export const HorizontalSlider = ({ title, movies }: HorizontalSliderProps) => {
    const containerHeight = (title) ? 260 : 220;
    return (
        <View style={{ ...styles.sliderContainer, height: containerHeight }} >
            {title && <Text style={styles.sliderText}>{title}</Text>}
            <FlatList
                horizontal
                keyExtractor={(item) => item.id.toString()}
                data={movies}
                renderItem={({ item }: any) => (
                    <MoviePoster movie={item} height={200} width={140} />
                )}
                showsHorizontalScrollIndicator={false}
            />
        </View>
    );
};

const styles = StyleSheet.create({
    sliderContainer: {
    },
    sliderText: {
        fontSize: 30,
        marginBottom: 10,
        color: 'white',
        fontWeight: 'bold',
        marginLeft: 10,
    },
});
