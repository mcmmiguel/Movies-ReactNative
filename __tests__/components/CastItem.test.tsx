import React from 'react';
import { screen, render } from '@testing-library/react-native';
import { CastItem } from '../../src/components/CastItem';
import { Cast } from '../../src/interfaces';

describe('Tests in <CastItem />', () => {

    const actor: Cast = {
        'adult': false,
        'gender': 2,
        'id': 3926,
        'known_for_department': 'Acting',
        'name': 'Albert Finney',
        'original_name': 'Albert Finney',
        'popularity': 9.731,
        'profile_path': '/8WCFO9lMHel1bHVYl5lj8pYEw2s.jpg',
        'cast_id': 1,
        'character': 'George Dunlap',
        'credit_id': '52fe4842c3a36847f815e705',
        'order': 0,
    };

    test('Should match with the snapshot', () => {

        render(<CastItem actor={actor} />);
        expect(screen).toMatchSnapshot();

    });

    test('Should render the data successfully', () => {

        render(<CastItem actor={actor} />);

        const img = screen.getByTestId('actor-image');
        const imgSource = img.props.source;
        const uri = `https://image.tmdb.org/t/p/w500${actor.profile_path}`;

        expect(img).toBeTruthy();
        expect(imgSource.uri).toBe(uri);
        expect(screen.getByText(actor.name)).toBeTruthy();
        expect(screen.getByText(actor.original_name)).toBeTruthy();
        screen.debug();

    });

});
