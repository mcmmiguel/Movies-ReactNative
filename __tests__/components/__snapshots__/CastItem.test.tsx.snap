// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <CastItem /> Should match with the snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#E1E5F2",
      "borderRadius": 10,
      "elevation": 9,
      "flexDirection": "row",
      "height": 60,
      "marginRight": 20,
      "paddingRight": 10,
      "paddingVertical": 10,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 3,
        "width": 0,
      },
      "shadowOpacity": 0.29,
      "shadowRadius": 4.65,
    }
  }
>
  <Image
    source={
      {
        "uri": "https://image.tmdb.org/t/p/w500/8WCFO9lMHel1bHVYl5lj8pYEw2s.jpg",
      }
    }
    style={
      {
        "borderRadius": 10,
        "height": 60,
        "resizeMode": "contain",
        "width": 60,
      }
    }
    testID="actor-image"
  />
  <View
    style={
      {
        "justifyContent": "center",
        "marginLeft": 10,
        "marginVertical": 4,
      }
    }
  >
    <Text
      style={
        {
          "color": "black",
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
    >
      Albert Finney
    </Text>
    <Text
      style={
        {
          "color": "black",
          "fontSize": 14,
        }
      }
    >
      George Dunlap
    </Text>
  </View>
</View>
`;
